examples/hybrid_jobs/6_QNSPSA_optimizer_with_embedded_simulator/qnspsa_with_embedded_simulator.ipynb

```diff
@@ -0,0 +1,1360 @@
+    "\n",
+    "For more detailed information and quick hands-on experience, refer to [this example notebook](https://github.com/aws/amazon-braket-examples/blob/main/examples/hybrid_jobs/4_Embedded_simulators_in_Braket_Jobs/Embedded_simulators_in_Braket_Jobs.ipynb)."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "11de3d4b",
+   "metadata": {},
+   "source": [
+    "## Background: QN-SPSA optimizer\n",
+    "\n",
+    "In an hand-waving argument, quantum natural simultaneous perturbation stochastic approximation (QN-SPSA, proposed by Gacon et al. [1]) is a second-order SPSA algorithm. \n",
+    "\n",
+    "The update rule of a vanilla gradient descent (GD) is written as:\n",
+    "$x_{t + 1} = x_{t} - \\eta \\nabla f_t \\label{eq:vanilla}\\tag{1}$ \n",
+    "\n",
+    "For quantum circuits, the gradient $\\nabla f_t$ is then estimated dimension by dimension, requiring $O(d)$ quantum measurements ($d$ being the dimension of the parameter space). As quantum measurements are expensive, this scaling makes GD impractical when dealing with high-dimensional complicated circuits.\n",
+    "\n",
+    "SPSA replaces this dimension-wise gradient estimation with a stochastic one [2]. In SPSA, a random sampled direction is used to provide a finite-difference approximation on the gradient. Although this stochasitic approach cannot provide a step-wise unbiased gradient estimation, SPSA is proved to be especially effective when accumulated over multiple optimization steps.\n",
+    "\n",
+    "On the other hand, quantum natural gradient descent (QNG) is a variant of gradient descent. It introduces the Fubini-Study metric tensor $F$ [4] into the optimization to account for the structure of the non-Euclidean parameter space [5]. With the metric tensor, the update rule is rewritten as:\n",
+    "\n",
+    "$x_{t + 1} = x_{t} - \\eta F^{-1}\\nabla f_t \\label{eq:qn}\\tag{2}$ \n",
+    "While the introduction of the metric tensor helps the optimization [3], the algorithm is not as scalable due to the number of measurements required to estimate $F$.\n",
+    "\n",
+    "QN-SPSA manages to combine the merits of both QNG and SPSA by estimating both the gradient and the metric tensor stochastically. The gradient is estimated in the same fashion as the SPSA algorithm, while the Fubini-Study metric is computed by a second-order process with another two stochastic perturbations. This $O(1)$ update rule could fit into NISQ devices well. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "efaeacf8",
+   "metadata": {},
+   "source": [
+    "## Implementation and notebook test\n",
+    "I start with implementing the QN-SPSA algorithm using Pennylane, and this prototype is done in the Braket notebook environment. I find the notebook a quite handy tool for this purpose. It is deployed with the necessary packages, saving me time to set up the environment on my local device. \n",
```

Comment:
I'm not sure we have to keep this paragraph; by the time users get to this notebook example they likely will already have an environment set up, whether locally or on a Braket NBI. We can use say that we start on a small example as a demonstration.

Reply:
My preference is to keep the discussion. Other than showcasing the research capability of the service, I would also like to use this notebook to recommend the practice of developing with NBI->local Jobs->remote Jobs. I personally think that we should promote the NBI more than we do right now, as developing in the cloud could really be the trend for ML/QML fields.